desktop/apps/consignments2/client/actions.js

```diff
@@ -153,31 +185,36 @@ export function showNotConsigningMessage () {
 
 export function submitDescribeWork (values) {
   return (dispatch) => {
-    dispatch(updateInputs(values)) // update the inputs in case we have to return
+    dispatch(updateInputs(values)) // update the inputs
+    dispatch(scrubLocation())
     dispatch(createSubmission()) // create the submission in convection
   }
 }
 
 export function submitPhoto () {
   return async (dispatch, getState) => {
     try {
+      dispatch(clearError())
       const { body: { token } } = await request
                             .post(`${sd.API_URL}/api/v1/me/token`)
                             .send({ client_application_id: sd.CONVECTION_APP_ID })
                             .set('X-ACCESS-TOKEN', sd.CURRENT_USER.accessToken)
       const {
         submissionFlow: {
-          submission
+          submission,
+          submissionIdFromServer
         }
       } = getState()
 
+      const submissionQueryParam = submission ? submission.id : submissionIdFromServer
       const { body } = await request
-                          .put(`${sd.CONVECTION_APP_URL}/api/submissions/${submission.id}`)
+                          .put(`${sd.CONVECTION_APP_URL}/api/submissions/${submissionQueryParam}`)
                           .set('Authorization', `Bearer ${token}`)
                           .send({ state: 'submitted' })
       dispatch(updateSubmission(body))
       dispatch(push('/consign2/submission/thank_you'))
     } catch (err) {
+      dispatch(updateError('Unable to submit at this time.'))
```

Comment:
I'll chat with @katarinabatina about what we actually want to say in the error cases 😄